[Bug]: `@storybook/theming/create` export no longer exists with 8.2.2

### Describe the bug

I'm upgrading our Storybook from 8.1 to 8.2.2 and ran into the error below:

```
1:17:48 PM [vite] Internal server error: Failed to resolve import "@storybook/theming/create" from ".storybook/sl-theme.ts". Does the file exist?
```

Code in question: `import { create } from '@storybook/theming/create';`

Afaict this code should still be valid (https://storybook.js.org/docs/configure/user-interface/theming#create-a-theme-quickstart)

If i inspect the theming package in my node_modules folder, i don't see anything anymore that relates to `create`.

### Reproduction link

-

### Reproduction steps

_No response_

### System

```bash
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Max
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.15.1 - /opt/homebrew/bin/node
    Yarn: 4.3.1 - /opt/homebrew/bin/yarn <----- active
    npm: 10.7.0 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 126.0.6478.127
    Safari: 17.5
  npmPackages:
    @storybook/addon-a11y: ^8.2.2 => 8.2.2 
    @storybook/addon-essentials: ^8.2.2 => 8.2.2 
    @storybook/addon-storysource: ^8.2.2 => 8.2.2 
    @storybook/core-events: ^8.2.2 => 8.2.2 
    @storybook/manager-api: ^8.2.2 => 8.2.2 
    @storybook/preview-api: ^8.2.2 => 8.2.2 
    @storybook/test: ^8.2.2 => 8.2.2 
    @storybook/theming: ^8.2.2 => 8.2.2 
    @storybook/web-components-vite: ^8.2.2 => 8.2.2 
    chromatic: ^11.5.5 => 11.5.5 
    storybook: ^8.2.2 => 8.2.2
```


### Additional context

_No response_

I see.. the issue you seem to get is this:
```
@orbitkit/storybook:build
cache miss, executing 63e832f9b940368b
$ tsc -p tsconfig.build.json
src/preview.tsx:49:7 - error TS4023: Exported variable 'preview' has or is using name 'ThemeVars' from external module "/home/runner/work/orbitkit/orbitkit/node_modules/@storybook/core/dist/theming/create" but cannot be named.

49 const preview = {
         ~~~~~~~

src/theme.ts:4:1 - error TS4082: Default export of the module has or is using private name 'ThemeVars'.

  4 export default create({
    ~~~~~~~~~~~~~~~~~~~~~~~
  5   base: 'dark',
    ~~~~~~~~~~~~~~~
... 
 27   textInverseColor: grayDark.gray1,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 28 })
 ```
 
 @ixahmedxi?
 
Is this a problem with our types?

I've seen the "but cannot be named" problem often too, and always end up here:
https://stackoverflow.com/questions/43900035/ts4023-exported-variable-x-has-or-is-using-name-y-from-external-module-but

We do export the `ThemeVars` type here:
https://github.com/storybookjs/storybook/blob/4ba799f0e44ef6c0365a3b306f8ffe2ee2db4ace/code/core/src/theming/types.ts#L5
re-exported here:
https://github.com/storybookjs/storybook/blob/c6472602bcc3bbb654653037c62bd87495bfb153/code/core/src/theming/index.ts#L32

So I'm not sure there's anything on the storybook side we could do, but I'm very open to suggests and would love to fix this so no changes are required on your part.

> I think there's something slightly different going on with PNPM workspaces as well.
> 
> Getting the same `create not exported` error, but I have `@storybook/theming` installed in this package.

We also have an issue in a pnpm workspace when upgrading to 8.2:
`Cannot find module 'storybook/internal/preview-api' from '../../node_modules/@storybook/react/dist/index.js'`

> I see.. the issue you seem to get is this:
> 
> ```
> @orbitkit/storybook:build
> cache miss, executing 63e832f9b940368b
> $ tsc -p tsconfig.build.json
> src/preview.tsx:49:7 - error TS4023: Exported variable 'preview' has or is using name 'ThemeVars' from external module "/home/runner/work/orbitkit/orbitkit/node_modules/@storybook/core/dist/theming/create" but cannot be named.
> 
> 49 const preview = {
>          ~~~~~~~
> 
> src/theme.ts:4:1 - error TS4082: Default export of the module has or is using private name 'ThemeVars'.
> 
>   4 export default create({
>     ~~~~~~~~~~~~~~~~~~~~~~~
>   5   base: 'dark',
>     ~~~~~~~~~~~~~~~
> ... 
>  27   textInverseColor: grayDark.gray1,
>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>  28 })
> ```
> 
> @ixahmedxi?
> 
> Is this a problem with our types?
> 
> I've seen the "but cannot be named" problem often too, and always end up here: https://stackoverflow.com/questions/43900035/ts4023-exported-variable-x-has-or-is-using-name-y-from-external-module-but
> 
> We do export the `ThemeVars` type here:
> 
> https://github.com/storybookjs/storybook/blob/4ba799f0e44ef6c0365a3b306f8ffe2ee2db4ace/code/core/src/theming/types.ts#L5
> 
> 
> re-exported here:
> https://github.com/storybookjs/storybook/blob/c6472602bcc3bbb654653037c62bd87495bfb153/code/core/src/theming/index.ts#L32
> 
> So I'm not sure there's anything on the storybook side we could do, but I'm very open to suggests and would love to fix this so no changes are required on your part.

I can't seem to be able to import `ThemeVars` from `@storybook/theming/create` but I can from `@storybook/core/theming` or `@storybook/theming`.

The error actually goes away when I import the `create` function from `@storybook/core/theming` or `@storybook/theming` instead of `@storybook/theming/create`.

Same problems happen here.

`ERROR in ./node_modules/.pnpm/@storybook+manager-api@7.6.17_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/@storybook/manager-api/dist/index.mjs 17:0-51`

`Module not found: Error: Can't resolve '@storybook/theming/create' in '/Users/XX/YYY/frontend/node_modules/.pnpm/@storybook+manager-api@7.6.17_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/@storybook/manager-api/dist'`

--- a/code/addons/docs/src/preset.ts
+++ b/code/addons/docs/src/preset.ts
@@ -70,7 +70,7 @@ async function webpack(
    */
   const cliPath = require.resolve('storybook/package.json');
   const themingPath = join(cliPath, '..', 'core', 'theming', 'index.js');
-  const themingCreatePath = join(cliPath, 'core', 'theming', 'create.js');
+  const themingCreatePath = join(cliPath, '..', 'core', 'theming', 'create.js');
 
   const componentsPath = join(cliPath, '..', 'core', 'components', 'index.js');
   const blocksPath = dirname(require.resolve('@storybook/blocks/package.json'));

--- a/code/addons/docs/src/preset.ts
+++ b/code/addons/docs/src/preset.ts
@@ -68,11 +68,11 @@ async function webpack(
    *
    * In the future the `@storybook/theming` and `@storybook/components` can be removed, as they should be singletons in the future due to the peerDependency on `storybook` package.
    */
-  const cliPath = require.resolve('storybook/package.json');
-  const themingPath = join(cliPath, '..', 'core', 'theming', 'index.js');
-  const themingCreatePath = join(cliPath, '..', 'core', 'theming', 'create.js');
+  const cliPath = dirname(require.resolve('storybook/package.json'));
+  const themingPath = join(cliPath, 'core', 'theming', 'index.js');
+  const themingCreatePath = join(cliPath, 'core', 'theming', 'create.js');
 
-  const componentsPath = join(cliPath, '..', 'core', 'components', 'index.js');
+  const componentsPath = join(cliPath, 'core', 'components', 'index.js');
   const blocksPath = dirname(require.resolve('@storybook/blocks/package.json'));
   if (Array.isArray(webpackConfig.resolve?.alias)) {
     alias = [...webpackConfig.resolve?.alias];

